### PR TITLE
feat: refactor miner contract

### PR DESCRIPTION
**Pull request by wenyuangui**
### Proposed changes in this pull request
1.refactor miner contract;
2.modify pov rpc api;

### Type
- [ ] Bug fix: (Link to the issue #{issue No.})
- [x] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### Checklist
Write a small comment explaining if its `N/A` (not applicable)

- [x] Read the [CONTRIBUTION](https://github.com/qlcchain/go-qlc/blob/master/CONTRIBUTING.md).
- [x] All the tests are passing after the introduction of new changes.
- [ ] Added tests respective to the part of code I have written.
- [x] Added proper documentation where ever applicable.
- [ ] Code has been written according to [Golang-Style-Guide](https://github.com/qlcchain/go-qlc/blob/master/CONTRIBUTING.md#code-standard)

### Extra information
Any extra information related to this pull request.

### CODE REVIEW

File: p2p/messageMonitor.go [hunk 1]
```diff
@@ -135,10 +135,7 @@ func (ms *MessageService) startLoop() {
 				}
 			default:
 				ms.netService.node.logger.Error("Received unknown message.")
-				time.Sleep(5 * time.Millisecond)
```
Comment (gythialy) [inline p2p/messageMonitor.go]: better to keep this sleep, when a large number of forged (unknown) messages are received, it maybe cost high cpu loading

File: p2p/messageMonitor.go [hunk 2]
```diff
@@ -206,11 +191,7 @@ func (ms *MessageService) confirmAckLoop() {
 			switch message.MessageType() {
 			case ConfirmAck:
 				ms.onConfirmAck(message)
-			default:
```
Comment (gythialy) [inline p2p/messageMonitor.go]: same as before

File: p2p/messageMonitor.go (hunk 1 above)
Comment (wenyuangui) [reply p2p/messageMonitor.go]: all msgs from messageCh is using subscribe by (ms *MessageService) Start, unknown msgs can not send to messageCh.

**Comment by wenyuangui** [comment]
miner contract method data is diff from previous, MUST redeploy new pov testnet, delay merge code to master.

**Pull request merged**